src/libsyntax/print/pprust.rs

```diff
@@ -3008,20 +3005,19 @@ impl<'a> State<'a> {
             Some(cm) => cm,
             _ => return Ok(())
         };
-        match self.next_comment() {
-            Some(ref cmnt) => {
-                if (*cmnt).style != comments::Trailing { return Ok(()) }
-                let span_line = cm.lookup_char_pos(span.hi);
-                let comment_line = cm.lookup_char_pos((*cmnt).pos);
-                let mut next = (*cmnt).pos + BytePos(1);
-                match next_pos { None => (), Some(p) => next = p }
-                if span.hi < (*cmnt).pos && (*cmnt).pos < next &&
-                    span_line.line == comment_line.line {
-                        self.print_comment(cmnt)?;
-                        self.cur_cmnt_and_lit.cur_cmnt += 1;
-                    }
-            }
-            _ => ()
+        if let Some(ref cmnt) = self.next_comment() {
+            if (*cmnt).style != comments::Trailing { return Ok(()) }
+            let span_line = cm.lookup_char_pos(span.hi);
+            let comment_line = cm.lookup_char_pos((*cmnt).pos);
+            let mut next = (*cmnt).pos + BytePos(1);
+            if let Some(p) = next_pos {
+                next = p;
+            }
+            if span.hi < (*cmnt).pos && (*cmnt).pos < next &&
+                span_line.line == comment_line.line {
+                    self.print_comment(cmnt)?;
+                    self.cur_cmnt_and_lit.cur_cmnt += 1;
+                }
```

Comment:
@zackmdavis I believe the rule is "when breaking up an if-condition onto multiple lines, the condition continuation line gets a **three**-space indent (so that it lines up with the first line of the if-condition), and the body is indented four spaces as usual".

I'm not sure if this is documented anywhere, but I've seen it a lot in the Rust code base. I've also seen other styles for the if-condition, like
```rust
            if span.hi < (*cmnt).pos && (*cmnt).pos < next
                 && span_line.line == comment_line.line {
```
but the body should always be indented four spaces no matter what.